lib/pages/settings/service/request_provider.dart

```diff
@@ -0,0 +1,40 @@
+import 'dart:async';
+
+import 'package:acs_upb_mobile/generated/l10n.dart';
+import 'package:acs_upb_mobile/pages/settings/model/request.dart';
+import 'package:acs_upb_mobile/widgets/toast.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+import 'package:flutter/cupertino.dart';
+
+extension RequestExtension on Request {
+  Map<String, String> toData() {
+    final Map<String, String> data = {};
+
+    if (userId != null) data['addedBy'] = userId;
+    if (requestBody != null) data['rerequestBody'] = requestBody;
+
+    return data;
+  }
+}
+
+class RequestProvider with ChangeNotifier {
+  final Firestore _db = Firestore.instance;
+
+  Future<bool> addForm(Request request, BuildContext context) async {
```

Comment:
Make `context` a named parameter and check whether it's null before displaying a message.